src/server/scripts/Spells/spell_priest.cpp

```diff
@@ -2670,58 +2671,125 @@ class spell_pri_schism : public SpellScript
     }
 };
 
-// 231682 - Sanctuary
-// Triggered by 585 - Smite
-class spell_pri_sanctuary : public SpellScript
+// 585 - Smite
+class spell_pri_smite : public SpellScript
 {
     bool Validate(SpellInfo const* /*spellInfo*/) override
     {
         return ValidateSpellInfo
         ({
-            SPELL_PRIEST_SMITE,
-            SPELL_PRIEST_SANCTUARY_AURA
+            SPELL_PRIEST_SANCTUARY,
+            SPELL_PRIEST_SANCTUARY_AURA,
+            SPELL_PRIEST_SANCTUARY_ABSORB
         });
     }
 
-    void HandleHit() const
+    void HandleEffectHit(SpellEffIndex /*effIndex*/) const
     {
-        if (GetCaster()->HasAura(SPELL_PRIEST_SANCTUARY))
-            GetCaster()->CastSpell(GetHitUnit(), SPELL_PRIEST_SANCTUARY_AURA, true);
+        Unit* caster = GetCaster();
+
+        if (caster->HasAura(SPELL_PRIEST_SANCTUARY))
+        {
+            caster->CastSpell(GetHitUnit(), SPELL_PRIEST_SANCTUARY_AURA, true);
+            caster->CastSpell(caster, SPELL_PRIEST_SANCTUARY_ABSORB, true);
```

Comment:
Why are you choosing caster as target? The one holding the aura should be the target of smite.

Reply:
from sniffs it looks like both target and caster get an aura, 208771 on caster(like a hidden one, from which the absorb happens) and 208772 on target